pkg/errors/error.go

```diff
@@ -24,51 +25,21 @@ func AbortWithError(ctx *gin.Context, code int, err error, details string) {
 	})
 }
 
-func (e *Error) Error() string {
-	return fmt.Sprintf("%s:%s", e.Message, e.Details)
-}
-
-func NewForbiddenError(message, details string) *Error {
-	return &Error{
-		Status:  http.StatusText(http.StatusForbidden),
-		Code:    http.StatusForbidden,
-		Message: message,
-		Details: details,
+// AbortWithStatusError stops the chain, writes the status code and the given error
+func AbortWithStatusError(ctx *gin.Context, err error, details string) {
+	code := http.StatusInternalServerError
+	if statusError, ok := err.(*errors2.StatusError); ok {
+		code = int(statusError.Status().Code)
 	}
-}
 
-func NewTooManyRequestsError(message, details string) *Error {
-	return &Error{
-		Status:  http.StatusText(http.StatusTooManyRequests),
-		Code:    http.StatusTooManyRequests,
-		Message: message,
-		Details: details,
-	}
-}
-
-func NewInternalError(err error, details string) *Error {
-	return &Error{
-		Status:  http.StatusText(http.StatusInternalServerError),
-		Code:    http.StatusInternalServerError,
-		Message: err.Error(),
-		Details: details,
-	}
-}
-
-func NewNotFoundError(err error, details string) *Error {
-	return &Error{
-		Status:  http.StatusText(http.StatusNotFound),
-		Code:    http.StatusNotFound,
+	ctx.AbortWithStatusJSON(code, &Error{
+		Status:  http.StatusText(code),
+		Code:    code,
 		Message: err.Error(),
 		Details: details,
```

Comment:
See https://projects.engineering.redhat.com/browse/RHDX-1403 for example

Reply:
@alexeykazakov Yeah.. i noticed that when i was reading the messages that were getting printed. I can reach out to the RHD folks and see if it would be a problem.